code/modules/mob/living/simple_animal/friendly/spiderbot.dm

```diff
@@ -0,0 +1,276 @@
+/mob/living/simple_animal/spiderbot
+	name = "Spider bot"
+	desc = "A skittering robotic friend!"
+	icon = 'icons/mob/robots.dmi'
+	icon_state = "spiderbot-chassis"
+	icon_living = "spiderbot-chassis"
+	icon_dead = "spiderbot-smashed"
+	language_holder = /datum/language_holder/universal
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 1, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 500
+	wander = FALSE
+	health = 10
+	maxHealth = 10
+	attacktext = "shocks"
+	melee_damage_lower = 2
+	melee_damage_upper = 2
+	response_help  = "pets"
+	response_disarm = "shoos"
+	response_harm   = "stomps on"
+	ventcrawler = 2
+	speed = -1                    //Spiderbots gotta go fast.
+	pass_flags = PASSTABLE
+	mob_size = MOB_SIZE_TINY
+	speak_emote = list("beeps","clicks","chirps")
+
+	var/obj/item/radio/borg/radio = null
+	var/obj/machinery/camera/camera = null
+	var/obj/item/mmi/mmi = null
+	var/list/req_access = list(ACCESS_ROBOTICS) //Access needed to pop out the brain.
+
+	var/emagged = 0
+	var/obj/item/held_item = null //Storage for single item they can hold.
+
+/mob/living/simple_animal/spiderbot/attackby(obj/item/O, mob/user)
+
+	if(istype(O, /obj/item/mmi))
+		var/obj/item/mmi/M = O
+		if(mmi) //There's already a brain in it.
+			to_chat(user, span_warning("There's already a brain in [src]!"))
+			return
+		if(!M.brainmob)
+			to_chat(user, span_warning("Sticking an empty MMI into the frame would sort of defeat the purpose."))
+			return
+		var/mob/living/brain/BM = M.brainmob
+		if(!BM.key || !BM.mind)
+			to_chat(user, span_warning("The MMI indicates that their mind is completely unresponsive; there's no point!"))
+			return
+
+		if(!BM.client) //braindead
+			to_chat(user, span_warning("The MMI indicates that their mind is currently inactive; it might change!"))
+			return
+
+		if(BM.stat == DEAD || BM.suiciding || (M.brain && (M.brain.brain_death || M.brain.suicided)))
+			to_chat(user, span_warning("Sticking a dead brain into the frame would sort of defeat the purpose!"))
+			return
+
+		if(is_banned_from(BM.ckey, "Cyborg") || QDELETED(src) || QDELETED(BM) || QDELETED(user) || QDELETED(M) || !Adjacent(user))
+			if(!QDELETED(M))
+				to_chat(user, span_warning("This [M.name] does not seem to fit!"))
+			return
+
+		if(!user.temporarilyRemoveItemFromInventory(M))
+			return
+
+		to_chat(user, span_notice("You install [M] in [src]!"))
+
+		transfer_personality(M)
+
+		update_icon()
```

Comment:
fairly certain you dont need update_icon at all it should just handle itself

Reply:
update_icon() is ok here